tests/prefetch_related/tests.py

```diff
@@ -708,6 +708,8 @@ def test_nested_prefetch_related_are_not_overwritten(self):
     def test_values_queryset(self):
         with self.assertRaisesMessage(ValueError, 'Prefetch querysets cannot use values().'):
             Prefetch('houses', House.objects.values('pk'))
+        # That error should not affect managers with custom ModelIterable subclasses
+        Prefetch('teachers', Teacher.objects_custom.all())
```

Comment:
I wonder if you might add an assertion that the `_iterable_class` for `Teacher.objects_custom.all()` is `ModelIterableSubclass` just to be sure that isn't inadvertently refactored away in some future change.